packages/@lwc/babel-plugin-component/src/decorators/index.js

```diff
@@ -216,6 +216,12 @@ function decorators({ types: t }) {
                 return;
             }
 
+            if (node.superClass === null) {
+                // Components *must* extend from either LightningElement or some other superclass (e.g. a mixin).
+                // We can skip classes without a superclass to avoid adding unnecessary registerDecorators() calls
```

Comment:
Just to confirm, we also don't need to call `registerDecorators()` or perform any of the subsequent logic on LightningElement, mixins, or other "superclasses" right?

Reply:
(I believe we do have decorators in mixins, but I'm not familiar enough with this area to know whether we need to process the mixin class itself or whether it'll be sufficient to process the class extending/using the mixin)

Reply:
Both superclasses and mixins may have fields, so we do need to call `registerDecorators()` on them, yes. For LightningElement no, we don't need to do that.

Luckily both superclasses and mixins extend from another class. I'll add a Karma test for this, though, since I realized we don't seem to have one for mixins.

Reply:
Great! Just a nitpick then, but my impression from the comment was that superclasses/mixins were similar to LightningElement in being top-level classes with no supers. If _everything_ (components, mixins, etc) that has fields *must* extend from another class, can we make that clearer?

Reply:
Ah you're right. Yes we can change this comment (and a similar one elsewhere).